src/pip/_internal/pep425tags.py

```diff
@@ -115,18 +115,12 @@ def get_abi_tag():
                     lambda: hasattr(sys, 'gettotalrefcount'),
                     warn=(impl == 'cp')):
             d = 'd'
-        if get_flag('WITH_PYMALLOC',
-                    lambda: impl == 'cp',
-                    warn=(impl == 'cp' and
-                          sys.version_info < (3, 8))) \
-                and sys.version_info < (3, 8):
+        if sys.version_info < (3, 8) and get_flag(
+                'WITH_PYMALLOC', lambda: impl == 'cp', warn=(impl == 'cp')):
```

Comment:
I would do `is_cpython = impl == 'cp'` above and use it everywhere.

Reply:
I was planning on doing exactly this. :)